app/models/order.rb

```diff
@@ -12,6 +13,8 @@ class Order < ApplicationRecord
   validates :name, :address, :email, presence: true
   validates :pay_type, inclusion: pay_types.keys
 
+  scope :by_date, ->(from = Date.today, to = (Date.today + 1)) { Order.where("created_at >= ?", from).where("created_at < ?", to) }
```

Comment:
no need to write `Order` in this
Use single where
Make use of range instead

Reply:
can i make a class method instead? if yes. how is it different from a scope?

Reply:
Made changes

Reply:
Please answer the above question

Reply:
In class method if the query could not found any record then nil is return but in case of scope empty ActiveRecord Relation object is returned